src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -703,4 +835,241 @@ export class NetworkAreaDiagramViewer {
             converterStationElement.querySelector('polyline');
         polyline?.setAttribute('points', polylinePoints);
     }
+
+    private moveLoopEdgeGroup(
+        edges: SVGGraphicsElement[],
+        mousePosition: Point
+    ) {
+        edges.forEach((edge) => {
+            // get edge element
+            const edgeId = edge.getAttribute('svgid');
+            if (!edgeId) {
+                return;
+            }
+            const translation = new Point(
+                mousePosition.x - this.initialPosition.x,
+                mousePosition.y - this.initialPosition.y
+            );
+            const edgeNode: SVGGraphicsElement | null = this.moveSvgElement(
+                edgeId,
+                translation
+            );
+            if (!edgeNode) {
+                return;
+            }
+            // store edge angles, to use them for bus node redrawing
+            this.storeHalfEdgeAngle(
+                edgeNode,
+                edgeNode.id + '.1',
+                0,
+                DiagramUtils.getPathAngle
+            );
+            this.storeHalfEdgeAngle(
+                edgeNode,
+                edgeNode.id + '.2',
+                1,
+                DiagramUtils.getPathAngle
+            );
```

Comment:
This can be done later, but I just realize that it's at the cost of requesting again the edgeId. As a compromise readability / performance that should be ok, don't you think?

Reply:
Moved code used for computing and storing edge angles